extensions/transfer-functions/transfer-functions-core/src/test/java/org/eclipse/dataspaceconnector/transfer/functions/core/TransferFunctionsCoreLocalTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.transfer.functions.core;
+
+import org.eclipse.dataspaceconnector.junit.launcher.EdcExtension;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferWaitStrategy;
+import org.eclipse.dataspaceconnector.spi.transfer.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.types.domain.metadata.DataEntry;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.transfer.functions.spi.flow.local.LocalTransferFunction;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.eclipse.dataspaceconnector.transfer.functions.core.TransferFunctionsCoreServiceExtension.ENABLED_PROTOCOLS_KEY;
+import static org.eclipse.dataspaceconnector.transfer.functions.core.TransferFunctionsCoreServiceExtension.TRANSFER_TYPE;
+
+/**
+ * Verifies a local flow control.
+ */
+@ExtendWith(EdcExtension.class)
+public class TransferFunctionsCoreLocalTest {
+    private CountDownLatch latch;
+    private boolean invoked;
+
+    @Test
+    void verifyLocalFlowControllerInvoked(TransferProcessManager processManager) throws InterruptedException {
+
+        var dataEntry = DataEntry.Builder.newInstance().id("test123").build();
+
+        var dataRequest = DataRequest.Builder.newInstance()
+                .id(UUID.randomUUID().toString())
+                .protocol("ids")
+                .destinationType("foo")
+                .dataEntry(dataEntry)
+                .managedResources(false)
+                .dataDestination(DataAddress.Builder.newInstance().type("test-protocol1").build())
+                .connectorId("test").build();
+
+        processManager.initiateProviderRequest(dataRequest);
+
+        latch.await(1, TimeUnit.MINUTES);
+        Assertions.assertTrue(invoked);
```

Comment:
we can always assertTrue(latch.await(...)), which would save us the assertTrue(invoked.get()).
```suggestion
        assertThat(latch.await(1, TimeUnit.MINUTES)).isTrue();
```
Also, I think we should try and stick with one assertion framework (here: AssertJ)